app/jobs/create_sponsor_svg_job.rb

```diff
@@ -0,0 +1,49 @@
+class CreateSponsorSvgJob < ApplicationJob
+  queue_as :low
+
+  def perform(blob)
+    return unless blob.svg?
+    return if blob.metadata[:format] == ENUMS::IMAGE_FORMATS::SPONSOR
+
+    users = blob.attachments.where(record_type: "User").map(&:record)
+    sponsor_svg = SvgsController.render(partial: "/svgs/sponsors/default", locals: {logo: embedded_svg(blob).html_safe})
+    filename = "sponsor-#{blob.filename.base}.svg"
+    Tempfile.open(filename, Rails.root.join("tmp")) do |file|
+      file.write sponsor_svg
+      users.each do |user|
+        file.rewind
+        next unless AuthorizedUser.new(user).can_create_sponsor_svg?
```

Comment:
Currently the auth check fails for non admins. I wonder if we should simply let this run for everyone. Thoughts?

Reply:
Probably.. although we need to make sure that they can't swap out the creative mid-campaign. It still must be approved.